src/pages/Swap/index.tsx

```diff
@@ -707,8 +707,12 @@ export function Swap({
             </TraceEvent>
           ) : chainId && chainId !== connectedChainId ? (
             <ButtonPrimary
-              onClick={() => {
-                switchChain(connector, chainId)
+              onClick={async () => {
+                try {
+                  await switchChain(connector, chainId)
+                } catch (error) {
+                  // Ignore error, which keeps the user on the previous chain.
```

Comment:
This should be checking if its a rejection from the user before ignoring the error.
You can re-use the existing utility for swap errors to disambiguate user rejections.

Reply:
Even if the chain switch fails for other reasons, I think the way to handle it is the same? I talk about this in my PR description also.

Reply:
> I'd suggest to ticket that as part of the web team backlog and come back to it, as this is a good incremental improvement.

You should add a TODO referencing the ticket as a comment here.

Reply:
i agree we shouldn't start ignoring all errors here, just the user rejected ones